conans/client/loader.py

```diff
@@ -231,7 +231,7 @@ def parse_conanfile(conanfile_path, python_requires):
         module, filename = _parse_conanfile(conanfile_path)
         try:
             conanfile = _parse_module(module, filename)
-            conanfile.python_requires = py_requires
+            conanfile.python_requires = {it.ref.name: it for it in py_requires}
```

Comment:
I am ok with changing it to a dict, but then, declaring multiple python-requires with the same name should raise an Exception, displaying the conflict. Please implement that check.